### ISSUE TITLE
`jolokia` input writes garbage metrics when `name` == ""

### ISSUE THREAD

**Issue by nfirvine** [influxdata/telegraf#2769]
### Relevant telegraf.conf:

```
[[outputs.file]]
  files = ["stdout"]
  data_format = "influx"
[[inputs.jolokia]]
  context = '/jolokia/list'
[[inputs.jolokia.metrics]]
  attribute = "HeapMemoryUsage"
  mbean = "java.lang:type=Memory"
  name = "heap_memory_usage"
[[inputs.jolokia.servers]]
  host = "10.31.101.243"
  port = "9099"
```

### System info:

```
$ telegraf -version
Telegraf v1.2.1 (git: release-1.2 3b6ffb344e5c03c1595d862282a6823ecb438cff)
$ uname -ar
Linux prd-xm-kafka-xmatters-local-server1 3.10.0-327.10.1.el7.x86_64 #1 SMP Tue Feb 16 17:03:50 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
$ cat /etc/redhat-release
CentOS Linux release 7.2.1511 (Core)
```

### Steps to reproduce:

1. Run telegraf with a jolokia input lacking a `servers.name` field (or `""`)
2. Witness invalid format for metrics

### Expected behavior:

Either:
- the tag should be included with an empty value (not sure if this is valid line protocol or even makes sense)
- the tag should be excluded
- the `name` config should have a default value that's something other than `""`; maybe hostname

### Actual behavior:

```$ telegraf --config telegraf.conf
2017-05-05T21:23:38Z I! Starting Telegraf (version 1.2.1)
2017-05-05T21:23:38Z I! Loaded outputs: file
2017-05-05T21:23:38Z I! Loaded inputs: inputs.jolokia
2017-05-05T21:23:38Z I! Tags enabled: host=redacted
2017-05-05T21:23:38Z I! Agent Config: Interval:10s, Quiet:false, Hostname:"redacted", Flush Interval:10s
jolokia,jolokia_name=,jolokia_port=9099,jolokia_host=10.31.101.243,host=redacted
``` 
Notice `jolokia_name=,`, which befouls the whole metric.

**Comment by danielnelson** [comment]
We should remove the tag in this case.  It looks like we don't check if `server.Host` is set either though, so we should either return an error if it is unset or default it to `127.0.0.1`

**Comment by danielnelson** [comment]
@nfirvine Could you test with the jolokia2 plugin (Available in the nightlies or in the upcoming 1.5)?  We will probably not make any further updates to the original jolokia plugin.

**Comment by nfirvine** [comment]
This bug isn't affecting me any more TBH. I'm going to close it.
